builtin/providers/google/resource_container_cluster.go

```diff
@@ -444,7 +444,10 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("name", cluster.Name)
 	d.Set("zone", cluster.Zone)
-	d.Set("additional_zones", cluster.Locations)
+
+	if _, ok := d.GetOk("additional_zones"); ok {
+		d.Set("additional_zones", cluster.Locations)
```

Comment:
Alternatively, you can set "additional_zones" to be Computed. See #11631 for the same problem there.